api/src/main/java/com/flipkart/poseidon/ds/trie/KeyWrapper.java

```diff
@@ -0,0 +1,17 @@
+package com.flipkart.poseidon.ds.trie;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by shrey.garg on 18/07/16.
+ */
+public class KeyWrapper<K> {
+    public K key;
+    public boolean wildCard = false;
+    public boolean wildPath = false;
```

Comment:
We may call this greedyWildCard than wildPath (as trie is neutral and doesn't assume anything around urls, paths etc)